indico/modules/designer/client/js/index.js

```diff
@@ -424,9 +429,62 @@ import {$T} from 'indico/utils/i18n';
           selectedItem.y = unzoom(y > 0 ? y : 0);
         }
       },
+      back: function() {
+        if (div) {
+          var backmostItem = _.minBy(
+            Object.values(items).filter(
+              item => selectedItem.id !== item.id && itemsOverlap(selectedItem, item)
+            ),
+            'zIndex'
+          );
+          var zIndex = Math.min(selectedItem.zIndex, backmostItem.zIndex - 1);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+          normalizeZindex();
+        }
+      },
+      front: function() {
+        if (div) {
+          var foremostItem = _.maxBy(
+            Object.values(items).filter(
+              item => selectedItem.id !== item.id && itemsOverlap(selectedItem, item)
+            ),
+            'zIndex'
+          );
+          var zIndex = Math.max(selectedItem.zIndex, foremostItem.zIndex + 1);
+          div.css('zIndex', zIndex);
+          selectedItem.zIndex = zIndex;
+          normalizeZindex();
+        }
+      },
     }[direction]());
   }
 
+  function itemsOverlap(item1, item2) {
+    const {x: left1, y: top1} = item1;
+    const {x: left2, y: top2} = item2;
+
+    const right1 = left1 + item1.width;
+    const bottom1 = top1 + (item1.height || 22);
+    const right2 = left2 + item2.width;
+    const bottom2 = top2 + (item2.height || 22);
+
+    if (left1 >= right2 || left2 >= right1) {
+      return false;
+    }
+    return !(top1 >= bottom2 || top2 >= bottom1);
+  }
+
+  function normalizeZindex() {
+    var increament = 1;
+    Object.values(items)
+      .sort((item1, item2) => item1.zIndex - item2.zIndex)
+      .forEach(item => {
+        item.zIndex = increament + 9;
```

Comment:
"increment"

Reply:
Also, this can be rewritten as (untested):

```js
_.sortBy(Object.values(items), 'zIndex').forEach((item, n) => { item.zIndex = n + 10; });
```

Reply:
Yes, this works